File: php/features.php
<?php
declare(strict_types=1);

// SxpVpn SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SxpVpnFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SxpVpnBaseFeature();
            case "test":
                return new SxpVpnTestFeature();
            default:
                return new SxpVpnBaseFeature();
        }
    }
}

File: py/features.py
# SxpVpn SDK feature factory

from feature.base_feature import SxpVpnBaseFeature
from feature.test_feature import SxpVpnTestFeature


def _make_feature(name):
    features = {
        "base": lambda: SxpVpnBaseFeature(),
        "test": lambda: SxpVpnTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
